### PR TITLE
Added section for priority and preemption

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
Introduce priority and preemption in docs. It may have styling issues. This is copied from upstream documentation.

/cc @sjenning @aveshagarwal @derekwaynecarr @mburke5678 

@aveshagarwal @mburke5678 - Can you help review this?

Addresses #10761

### CODE REVIEW

**Comment by aveshagarwal** [comment]
I think adding info about default priority classes system-cluster-critical and system-node-critical would be helpful.

**Comment by ravisantoshgudimetla** [comment]
> I think adding info about default priority classes system-cluster-critical and system-node-critical would be helpful.

That makes sense. I will add it. Thanks :)

**Comment by aveshagarwal** [comment]
@ravisantoshgudimetla @mburke5678 the changes LGTM.

**Comment by mburke5678** [comment]
@ravisantoshgudimetla @aveshagarwal This PR is related to Trello: [Need documentation that ExperimentalCriticalPodAnnotation is being enabled by default and "scheduler.alpha.kubernetes.io/critical-pod": "" annotation is restricted to openshift-* projects.
](https://trello.com/c/ebBOFnRQ/908-1-schedulerneed-documentation-that-experimentalcriticalpodannotation-is-being-enabled-by-default-and-scheduleralphakubernetesio)

**Comment by mburke5678** [comment]
@mdshuai PTAL

**Comment by mburke5678** [comment]
@ravisantoshgudimetla I think we might need a rebase to fix the [Trello error above](https://travis-ci.org/openshift/openshift-docs/builds/402183553?utm_source=github_status&utm_medium=notification).

**Comment by ravisantoshgudimetla** [comment]
This PR is related to https://trello.com/c/TaS42RaD/911-3-consuming-priority-in-openshift

**Comment by aveshagarwal** [comment]
> @ravisantoshgudimetla @aveshagarwal This PR is related to Trello: Need documentation that ExperimentalCriticalPodAnnotation is being enabled by default and "scheduler.alpha.kubernetes.io/critical-pod": "" annotation is restricted to openshift-* projects.
??

@mburke5678 No. Its trello card is https://trello.com/c/TaS42RaD/911-3-consuming-priority-in-openshift

**Comment by mburke5678** [comment]
@ravisantoshgudimetla @aveshagarwal Is this for 3.10? The associated issue is titled 3.11 https://github.com/openshift/openshift-docs/issues/10761

**Comment by aveshagarwal** [comment]
> @ravisantoshgudimetla @aveshagarwal Is this for 3.10? The associated issue is titled 3.11 #10761

This is only for 3.11.

**Comment by wjiangjay** [comment]
@mburke5678 I have to check all of this days later since 3.11 cluster is still can not setup to make sure priority/preemption work well.

File: admin_guide/scheduling/priority_preemption.adoc [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Priority also affects scheduling order of Pods and out-of-resource eviction ordering on the Node.Pod priority and preemption .
+
+== How to use priority and preemption
+
+To use priority and preemption, follow these steps:
+
+. Add one or more PriorityClasses.
+
+. Create Pods withpriorityClassName set to one of the added PriorityClasses. Of course you do not need to create the Pods directly; normally you would add priorityClassName to the Pod template of a collection object like a Deployment.
```
Comment (wjiangjay) [inline admin_guide/scheduling/priority_preemption.adoc]: withpriorityClassName, need space after with here.

**Comment by wjiangjay** [comment]
I can not accept this PR since dev still not approve my test suites in trello https://trello.com/c/rraktdgG/348-13-scheduler-priority-preemption-scheduling

File: admin_guide/scheduling/priority_preemption.adoc [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+If you try the feature and then decide to disable it, you must remove the PodPriority command-line flag or set it to false, and then restart the API server and scheduler. After the feature is disabled, the existing Pods keep their priority fields, but preemption is disabled, and priority fields are ignored. If the feature is disabled, you cannot set priorityClassName in new Pods
+
+== How to use priority and preemption
+
+Preemption is controlled by a scheduler flag disablePreemption, which is set to false by default.
+
+To disable preemption, set disablePreemption to true. This keeps pod priority enabled but disables preemption. Here is a sample configuration:
+
+.Sample component config with preemption disabled
```
Comment (wjiangjay) [inline admin_guide/scheduling/priority_preemption.adoc]: I think OpenShift still not support this usage for v3.11, so need update here.

**Comment by vikram-redhat** [comment]
@mburke5678 - is this ready for updates after QE review?

**Comment by mburke5678** [comment]
@ravisantoshgudimetla Is this ready to move past QE review?

File: admin_guide/scheduling/priority_preemption.adoc (hunk 2 above)
Comment (ravisantoshgudimetla) [reply admin_guide/scheduling/priority_preemption.adoc]: We don't have a resource type called component config in API server, these are configs loaded from disk. We can pass `kube-scheduler --config=/path/to/config` but I think this section is confusing in general as we do have openshift master config file from which we are loading the information.

**Comment by ravisantoshgudimetla** [comment]
@wjiangjay Do you have any other comments, if not can you please approve it.

**Comment by mburke5678** [comment]
@ravisantoshgudimetla Did you remove the section with _Sample component config with preemption disabled_ that @wjiangjay was questioning? 
Can I merge this PR?

**Comment by ravisantoshgudimetla** [comment]
@mburke5678 

>  Did you remove the section with Sample component config with preemption disabled that @wjiangjay was questioning?

Yes, I removed it but not that it is wrong. I did that because it's confusing in OpenShift as there are multiple ways to achieve it. 

To me this PR, LGTM.

File: admin_guide/scheduling/priority_preemption.adoc [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+==== Cross node preemption
+
+Suppose a Node N is being considered for preemption so that a pending Pod P can be scheduled on N. P might become feasible on N only if a Pod on another Node is preempted. Here’s an example:
+
+. Pod P is being considered for Node N.
+. Pod Q is running on another Node in the same Zone as Node N.
+. Pod P has Zone-wide anti-affinity with Pod Q (topologyKey: failure-domain.beta.kubernetes.io/zone).
+. There are no other cases of anti-affinity between Pod P and other Pods in the Zone.
+. In order to schedule Pod P on Node N, Pod Q can be preempted, but scheduler does not perform cross-node preemption. So, Pod P will be deemed unschedulable on Node N.
```
Comment (mburke5678) [inline admin_guide/scheduling/priority_preemption.adoc]: @ravisantoshgudimetla Why does the scheduler not perform cross-node preemption? Are there specific situations where the scheduler will and won't?

File: admin_guide/scheduling/priority_preemption.adoc [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+Preemption is controlled by a scheduler flag disablePreemption, which is set to false by default.
+
+It is important to note that critical pods (except DaemonSet pods, which are still scheduled by the DaemonSet controller) rely on scheduler preemption to be scheduled when a cluster is under resource pressure. For this reason, we do not recommend disabling preemption.
+
+== PriorityClass
+
+A PriorityClass is a non-namespaced object that defines a mapping from a priority class name to the integer value of the priority. The name is specified in the name field of the PriorityClass object’s metadata. The value is specified in the required value field. The higher the value, the higher the priority.
+
+A PriorityClass object can have any 32-bit integer value smaller than or equal to 1 billion. Larger numbers are reserved for critical system Pods that should not normally be preempted or evicted. We have two reserved priorityClasses for critical system Pods to have guaranteed scheduling.
+
```
Comment (mburke5678) [inline admin_guide/scheduling/priority_preemption.adoc]: @ravisantoshgudimetla Does the user need to add one of these priority classes to critical pods? Is it done automatically? For new critical pods and upgrade scenarios.

**Pull request merged**